### PR TITLE
Analyzer added indexes support

### PR DESCRIPTION
**Pull request by kitaisreal**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: src/Interpreters/ActionsDAG.cpp [hunk 1]
```diff
@@ -2050,4 +2050,122 @@ bool ActionsDAG::isSortingPreserved(
...
+    auto & result_dag_outputs = result_dag->getOutputs();
+    result_dag_outputs.reserve(filter_nodes_size);
+
+    for (const auto & node : filter_nodes)
+        result_dag_outputs.push_back(node_to_result_node.find(node)->second);
+
+    if (result_dag_outputs.size() > 1)
+    {
+        auto function_builder = FunctionFactory::instance().get("and", context);
+        result_dag_outputs = {&result_dag->addFunction(function_builder, result_dag_outputs, {})};
```
Comment (novikd) [inline src/Interpreters/ActionsDAG.cpp:2165]:
```suggestion
        result_dag_outputs = { &result_dag->addFunction(function_builder, result_dag_outputs, {}) };
```

File: src/Planner/TableExpressionData.h [hunk 2]
```diff
@@ -91,16 +91,22 @@ class TableExpressionData
...
-    /// Get columns
-    const NamesAndTypesList & getColumns() const
+    /// Get column names
+    Names getColumnNames() const
     {
-        return columns;
+        Names result;
+        result.reserve(column_name_to_column.size());
+
+        for (const auto & [column_name, column] : column_name_to_column)
```
Comment (novikd) [inline src/Planner/TableExpressionData.h:106]:
```suggestion
        for (const auto & [column_name, _] : column_name_to_column)
```

File: src/Planner/TableExpressionData.h [hunk 3]
```diff
@@ -115,6 +121,32 @@ class TableExpressionData
...
 
+    /** Get column for column name.
+      * Exception is thrown if there are no column for column name.
+      */
+    const NameAndTypePair & getColumnOrThrow(const std::string & column_name) const
+    {
+        auto it = column_name_to_column.find(column_name);
+        if (it == column_name_to_column.end())
+            throw Exception(ErrorCodes::LOGICAL_ERROR,
+                "Column for name {} does not exists",
```
Comment (novikd) [inline src/Planner/TableExpressionData.h:132]: Maybe also add information about existing columns from `getColumnNames()`?

File: src/Processors/QueryPlan/ReadFromMergeTree.cpp [hunk 4]
```diff
@@ -1380,7 +1440,7 @@ void ReadFromMergeTree::describeActions(JSONBuilder::JSONMap & map) const
 void ReadFromMergeTree::describeIndexes(FormatSettings & format_settings) const
 {
     auto result = getAnalysisResult();
-    auto index_stats = std::move(result.index_stats);
+    auto index_stats = result.index_stats;
```
Comment (novikd) [inline src/Processors/QueryPlan/ReadFromMergeTree.cpp:1443]: Why do you copy it? Maybe `auto & index_stats` is better?

**Review by novikd — APPROVED** [review]
LGTM

File: src/Interpreters/ActionsDAG.cpp (hunk 1 above)
Comment (kitaisreal) [reply src/Interpreters/ActionsDAG.cpp:2165]: Fixed.

File: src/Planner/TableExpressionData.h (hunk 3 above)
Comment (kitaisreal) [reply src/Planner/TableExpressionData.h:132]: Updated.

File: src/Planner/TableExpressionData.h (hunk 2 above)
Comment (kitaisreal) [reply src/Planner/TableExpressionData.h:106]: Updated.

**Pull request merged**